src/controller/c/supervisor.c

```diff
@@ -979,12 +1091,14 @@ static void supervisor_read_answer(WbDevice *d, WbRequest *r) {
       // field_type == 0 if node was deleted
       if (field_type != 0) {
         const int field_node_id = request_read_int32(r);
+        const int field_proto_id = request_read_int32(r);
         const int field_id = request_read_int32(r);
         const bool is_field_get_request = sent_field_get_request && sent_field_get_request->field &&
                                           sent_field_get_request->field->node_unique_id == field_node_id &&
+                                          sent_field_get_request->field->proto_id == field_proto_id &&
                                           sent_field_get_request->field->id == field_id;
-        WbFieldStruct *f =
-          (is_field_get_request) ? sent_field_get_request->field : find_field_by_id(field_node_id, field_id, false);
+        WbFieldStruct *f = (is_field_get_request) ? sent_field_get_request->field :
+                                                    find_field_by_id(field_node_id, -1, field_id, false);  // TODO: FIX
```

Comment:
Is there still something to fix here or is the TODO no longer valid?

Reply:
Ahh, good catch! -1 should be replaced with the actual proto id. Fixed.